semgrep/semgrep/output.py

```diff
@@ -105,22 +85,39 @@ def build_normal_output(
                     severity_prepend = f"{YELLOW_COLOR}{severity} "
                 else:
                     severity_prepend = f"{severity} "
-            yield f"{severity_prepend}{YELLOW_COLOR}rule:{check_id}: {finding.get('extra', {}).get('message')}{RESET_COLOR}"
+            yield f"{severity_prepend}{YELLOW_COLOR}rule:{check_id}: {message}{RESET_COLOR}"
 
         last_file = current_file
         last_message = message
-        yield from finding_to_line(finding, color_output)
+        yield from finding_to_line(rule_match, color_output)
         if fix:
             yield f"{BLUE_COLOR}autofix:{RESET_COLOR} {fix}"
 
 
-def build_output_json(output_json: Dict[str, Any]) -> str:
+def build_output_json(rule_matches: List[RuleMatch], semgrep_errors: List[Any]) -> str:
     # wrap errors under "data" entry to be compatible with
     # https://docs.r2c.dev/en/latest/api/output.html#errors
-    errors = output_json["errors"]
-    if errors:
-        output_json["errors"] = {
-            "data": output_json["errors"],
-            "message": "SgrepRuntimeErrors",
-        }
+    output_json = {}
+    output_json["results"] = [rm.to_json() for rm in rule_matches]
+
+    errors = []
+    for error in semgrep_errors:
+        errors.append(
+            {"data": semgrep_errors, "message": "SemgrepCoreRuntimeErrors",}
+        )
+    output_json["errors"] = errors
     return json.dumps(output_json)
+
+
+def build_output(
+    rule_matches: List[RuleMatch],
+    semgrep_errors: List[Any],
+    json_format: bool,
+    color_output: bool,
+) -> str:
+    if json_format:
+        return build_output_json(rule_matches, semgrep_errors)
+    else:
+        return "\n".join(
+            build_normal_output(rule_matches, semgrep_errors, color_output)
```

Comment:
we could make this return a string instead of a list? idk maybe we use the list in other places

Reply:
Oh you're right I don't think it is called anywhere that uses it as a list. Will change.

Reply:
Oh build_normal_output uses yielding that I probably didn't want to deal with refactoring to a single return in this PR